app/models/subscription.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class Subscription < ApplicationRecord
+  validates :duration, presence: true, numericality: { only_integer: true, greater_than: 0 }
+  validate :cannot_change_after_cancelled, :cannot_change_duration, on: :update
```

Comment:
Should we also validate the date of cancelled_at? (for instance when set, it should not be in the past)

Reply:
Let's not ^^

Reply:
No specific benefits from that